macros/src/diagramgen.rs

```diff
@@ -1,10 +1,14 @@
+use std::collections::btree_set::Difference;
+
 use crate::parser::*;
 
 /// Generates a string containing 'dot' syntax to generate a statemachine diagram with graphviz.
 pub fn generate_diagram(sm: &ParsedStateMachine) -> String {
     let transitions = &sm.states_events_mapping;
 
-    let diagram_states = sm.states.iter().map(|s| s.0);
+    let mut diagram_states  = sm.states.iter().map(|s| s.0).collect::<Vec<&String>>();
+    diagram_states.sort();
+    let diagram_states = diagram_states.into_iter();
```

Comment:
What's the purpose of these various sorting changes?

Reply:
Diagramgen (and all smlang) uses HashMaps as it's data structure, which does not guarantee any order of elements. 

Therefore sorting is needed to ensure stable (ie not changing between runs of the same sm code) dot file contents.  This is needed to ensure stable hash sum, which is used to name unnamed diagrams. If done without sorting, the output is polluted with lots of similar svg files with different names.
This ensures that new files will only occur upon changing the structure of the code.
Maybe we should somehow manage the created files?

Reply:
Since this is build time, it's fine. Can you add this info as comments into the code for posterity though? Seems like useful intel :)